packages/toolpad-app/src/toolpad/propertyControls/ChartData.tsx

```diff
@@ -0,0 +1,247 @@
+import * as React from 'react';
+import {
+  Box,
+  Button,
+  IconButton,
+  List,
+  ListItem,
+  ListItemButton,
+  ListItemText,
+  MenuItem,
+  Popover,
+  Stack,
+  TextField,
+  Typography,
+} from '@mui/material';
+import AddIcon from '@mui/icons-material/Add';
+import { CHART_DATA_SERIES_KINDS, type ChartDataSeries } from '@mui/toolpad-components';
+import { useBrowserJsRuntime } from '@mui/toolpad-core/jsBrowserRuntime';
+import DeleteIcon from '@mui/icons-material/Delete';
+import { BindableAttrValue } from '@mui/toolpad-core';
+import * as appDom from '../../appDom';
+import type { EditorProps } from '../../types';
+import PropertyControl from '../../components/PropertyControl';
+import { usePageEditorState } from '../AppEditor/PageEditor/PageEditorProvider';
+import { useDom, useDomApi } from '../AppState';
+// eslint-disable-next-line import/no-cycle
+import BindableEditor from '../AppEditor/PageEditor/BindableEditor';
```

Comment:
I wonder whether we can solve all these cycles by creating a react context very high up in the toolpad app that holds a registry of all available propertyControls.

Reply:
Yeah it's becoming a bigger issue when we add controls with bindable props, I was focused on functionality first but will see if I can find a way to fix this.

Reply:
Turns out the cyclical dependency with prop controls was simpler than the other one with the binding editors - so passing the prop controls as context does fix the issue, nice one! I had thought it was part of the same problem.

I put the list of prop controls to pass to the context in the `ComponentEditor` component - let me know if you think this should be done differently in terms of file structure.